src/Microsoft.DocAsCode.MarkdigEngine.Extensions/ChromelessForms/ChromelessFormRender.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.MarkdigEngine.Extensions
+{
+	using System;
+
+	using Markdig.Renderers;
+	using Markdig.Renderers.Html;
+
+	public class ChromelessFormRender : HtmlObjectRenderer<ChromelessFormBlock>
+	{
+		protected override void Write(HtmlRenderer renderer, ChromelessFormBlock obj)
+		{
+			obj.HtmlAttributes.AddClass("chromeless-form"); ;
+			renderer.Write("<form")
+				.WriteAttributes(obj)
+				.Write(@" class=""chromeless-form""")
+				.Write($"{(String.IsNullOrEmpty(obj.Model) ? "" : $" data-model=\"{obj.Model}\"")}")
+				.Write($"{(String.IsNullOrEmpty(obj.Action) ? "" : $" data-action=\"{obj.Action}\"")}")
```

Comment:
escape `obj.Model`/`obj.Action` or put it to `obj.HtmlAttributes`

Reply:
@yufeih I actually attempted this, but I couldn't get the output to work using the `WriteAttributes(obj)` call, which is why I added this afterwards (along with the `class` declaration, which also should be handled through `HtmlAttributes`. I looked at the Markdig codebase on Github and it appears that `WriteAttributes` uses reflection to get the `HtmlAttributes.Properties` list of KeyValuePairs and write it to the output. (You may be able to see this in the initial commit, actually.) I wasn't seeing the output at all. Is there something that I may have missed?

Reply:
Markdig doesn't create the HtmlAttributes instance by default. I believe they do this to avoid the unnecessary memory overhead.

In triple-colon, I allow the extension implementations to return an `HtmlAttributes` collection via `TryProcessAttributes`. If an instances is returned it is associated with the block like this:
https://github.com/dotnet/docfx/pull/2929/files#diff-ef494d7b545157d836bf679679191860R64

Reply:
@jdanyow  Calling `AddClass` or `AddProperty` adds the collection if it doesn't exist. See https://github.com/lunet-io/markdig/blob/master/src/Markdig/Renderers/Html/HtmlAttributes.cs. I tried calling this in the renderer and in the parser, but I never saw the results in the HTML output.

Reply:
you have to create an HtmlAttributes instance and associate it with your block.
```cs
var htmlAttributes = new HtmlAttributes();
block.SetData(typeof(HtmlAttributes), htmlAttributes);

// set an attribute
htmlAttributes.AddProperty("data-zone", target);
```

If it makes sense, consider rebasing on #2929

Then you'd just need to implement this interface: 
https://github.com/jdanyow/docfx/blob/ec4f54f169a85f153db6889caf2ba426a85a491f/src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/TripleColonExtension.cs#L50-L56

In `TryProcessAttributes` you'd validate the markdown attributes and transforms them to html attributes or class names as appropriate. You may need to extend the interface if you have custom rendering beyond html attributes and classes.

Reply:
@jdanyow See https://github.com/dotnet/docfx/pull/2911/files#diff-94c4df4127ff307a120f542c615c487bR20. Regardless, I'll rebase this to clean up the code.